3200-1_1.xml

```diff
@@ -0,0 +1,148 @@
+				<Name>Digital Input Polarity</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Boolean</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The polarity of the digital input as a Boolean (False = Normal, True = Reversed).</Description>
+			</Item>
+			<Item ID="5503">
+				<Name>Digital Input Debounce</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units>ms</Units>
+				<Description>The debounce period in ms.</Description>
+			</Item>
+			<Item ID="5504">
+				<Name>Digital Input Edge Selection</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration>1..3</RangeEnumeration>
+				<Units></Units>
+				<Description>The edge selection as an integer (1 = Falling edge, 2 = Rising edge, 3 = Both Rising and Falling edge).</Description>
+			</Item>
+			<Item ID="5505">
+				<Name>Digital Input Counter Reset</Name>
+				<Operations>E</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type></Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>Reset the Counter value.</Description>
+			</Item>
+			<Item ID="5750">
+				<Name>Application Type</Name>
+				<Operations>RW</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The application type of the sensor or actuator as a string, for instance, "Air Pressure"</Description>
+			</Item>
+			<Item ID="5751">
+				<Name>Sensor Type</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The type of the sensor (for instance PIR type)</Description>
+			</Item>
+			<Item ID="5518">
+				<Name>Timestamp</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Time</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description>The timestamp of when the measurement was performed.</Description>
+			</Item>
+			<Item ID="6050">
+				<Name>Fractional Timestamp</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Optional</Mandatory>
+				<Type>Float</Type>
+				<RangeEnumeration>0..1</RangeEnumeration>
+				<Units>s</Units>
+				<Description>Fractional part of the timestamp when sub-second precision is used (e.g., 0.23 for 230 ms).</Description>
```

Comment:
This is weird. I am assuming that the case is a lwm2m client sending the current time (e.g., 1582642022), by adding fractional time we add subsecond precission (e.g., 0.23 if it is 230 ms).
I think Fractional Time should be the current time in ms (e.g., 1582642022.23) and that it should be used instead of Timestamp. Or just use integer and say that it is time in ms.

There is also the issue that senml also does send timestamps, so you may end up with redundant information.

Reply:
Only fractional (floating point) time doesn't work well since not all platforms support float types. Also just integer (fixed point) does not work well since we would have to fix the precision (ms/us/ns?) in the resource definition. 

One potential alternative would be to create a new "Float timestamp" resource that could be used when float is supported, and could include the non-fractional part of the time too. But that should not be used together with the timestamp or the fractional part resources, or we run into weird error cases. Hence, that might not be worth the effort.

For SenML time stamp configuration a new attribute might be a good solution.

Reply:
> There is also the issue that senml also does send timestamps, so you may end up with redundant information.

This was already commented earlier by @dnav, so check my comment above. Since the LwM2M specification does not specify LwM2M server north bound interface, it depends on the server implementation if senml timestamps are available via nb interface for applications.

Reply:
Any reasonable LwM2M server implementation should expose all information available from the clients. I guess the problem here is that using SenML time stamps was never properly defined in the LwM2M TS. That could be fixed in future versions. 

But since we have timestamp functionality already in some IPSO SOs, I would not object making it uniformly available as in this PR.

Reply:
> Any reasonable LwM2M server implementation should expose all information available from the clients. I guess the problem here is that using SenML time stamps was never properly defined in the LwM2M TS. That could be fixed in future versions.

I fully agree. We have done quite extensive evaluation of available LwM2M server implementations and the absence of senml timestamp in nb interface is unfortunately common. We need to be able to sell our LwM2M capable devices also "as-is" without our own cloud application. It would be sad to say to our customers that the device is not compatible with your LwM2M server, since it does not provide sensor measurement timestamps from senml messages. Instead we can say that your LwM2M server is not fully supported, but you have the workaround to read or observe timestamp resources in sensor object instances.